external/sql/storm-sql-core/src/test/org/apache/storm/sql/compiler/TestCompilerUtils.java

```diff
@@ -153,6 +158,29 @@ public static CalciteState sqlOverNestedTable(String sql)
                 .field("MAPFIELD", SqlTypeName.ANY)
                 .field("NESTEDMAPFIELD", SqlTypeName.ANY)
                 .field("ARRAYFIELD", SqlTypeName.ANY)
+
+                // TODO: replace field definitions with belows while Calcite can handle properly
```

Comment:
From what I'm experimenting, nested map access seems supported, but its type should be ANY instead of MAP. I'll remove this TODO for now.